### PR TITLE
Bug 1886452: Fix for topology layout options style on select

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**
https://issues.redhat.com/browse/ODC-4980

**Description**
Update the current layout variable on layout change so the buttons draw selection appropriately.

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/kind bug

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx [hunk 1]
```diff
@@ -16,7 +16,14 @@ interface TopologyControlBarProps {
 }
 
 const TopologyControlBar: React.FC<TopologyControlBarProps> = ({ visualization }) => {
-  const layout = visualization.getGraph()?.getLayout() ?? 'COLA_LAYOUT';
+  const [layout, setLayout] = React.useState<string>(
+    visualization?.getGraph()?.getLayout() ?? 'COLA_LAYOUT',
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:20]: I expect you want to use the const here:

```suggestion
    visualization?.getGraph()?.getLayout() ?? COLA_LAYOUT,
```

Comment (jerolimov) [reply frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:20]: Oh see now it was previously also this string.. But I expect we should fix this anyway :smirk:

**Comment by jerolimov** [comment]
Tested it locally, works fine and solves the issue. If you can use the const above I would add a lgtm.

**Review by jerolimov — CHANGES_REQUESTED** [review]
Tested it locally, works fine and solves the issue. If you can use the const above I would add a lgtm.

**Comment by invincibleJai** [comment]
Looks good and works as expected!!

**Comment by jeff-phillips-18** [comment]
Updated. Thanks @jerolimov

File: frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx [hunk 2]
```diff
@@ -16,7 +16,14 @@ interface TopologyControlBarProps {
...
 
 const TopologyControlBar: React.FC<TopologyControlBarProps> = ({ visualization }) => {
-  const layout = visualization.getGraph()?.getLayout() ?? 'COLA_LAYOUT';
+  const [layout, setLayout] = React.useState<string>(
+    visualization?.getGraph()?.getLayout() ?? COLA_LAYOUT,
+  );
+  React.useEffect(() => {
+    visualization.getGraph().setLayout(layout);
+    visualization.getGraph().layout();
+  }, [layout, visualization]);
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:25]: Sorry that I missed this in the first round.. This effect was also called after the first rendering and it calls setLayout and layout right after the initial rendering. When seeing this I checked that setLayout [do nothing if the layoutType does not changed](https://github.com/patternfly/patternfly-react/blob/2c4b25ce8aa64a08de38c0a1884c96df643f348e/packages/react-topology/src/elements/BaseGraph.ts#L109-L111).

If the initial layout call here required? Should we ignore this or do we want do this only when the user pressed one of the buttons? :thinking:

Comment (christianvogt) [reply frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:25]: Ya we should avoid a call to `layout` here on first render.

Alternative to `useState` would be to just make the control bar an `observer` which will force re-render when layout changes using the old implementation.

Comment (jerolimov) [reply frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:25]: Or useState/useRef and call setLayout and layout again in an onClick handler.

Comment (jeff-phillips-18) [reply frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx]: Thanks. Just added the observer and updated to use the constant.

File: frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx [hunk 3]
```diff
@@ -15,8 +16,8 @@ interface TopologyControlBarProps {
   visualization: Visualization;
 }
 
-const TopologyControlBar: React.FC<TopologyControlBarProps> = ({ visualization }) => {
-  const layout = visualization.getGraph()?.getLayout() ?? 'COLA_LAYOUT';
+const TopologyControlBar: React.FC<TopologyControlBarProps> = observer(({ visualization }) => {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:19]: `observer` makes the component a `memo`. You can remove your call to memo below.

Comment (jeff-phillips-18) [reply frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:19]: Fixed. Thanks.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

File: frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx (hunk 3 above)
Comment (jerolimov) [reply frontend/packages/dev-console/src/components/topology/TopologyControlBar.tsx:19]: Hmm interesting.. reading the docs.. :)

**Pull request merged**